src/borg/archiver.py

```diff
@@ -751,20 +759,19 @@ def do_list(self, args, repository, manifest, key):
                     format = "{mode} {user:6} {group:6} {size:8} {isomtime} {path}{extra}{NL}"
                 formatter = ItemFormatter(archive, format)
 
-                if not hasattr(sys.stdout, 'buffer'):
-                    # This is a shim for supporting unit tests replacing sys.stdout with e.g. StringIO,
-                    # which doesn't have an underlying buffer (= lower file object).
-                    def write(bytestring):
-                        sys.stdout.write(bytestring.decode('utf-8', errors='replace'))
-                else:
-                    write = sys.stdout.buffer.write
                 for item in archive.iter_items(lambda item: matcher.match(item.path)):
                     write(safe_encode(formatter.format_item(item)))
         else:
+            archive_info_keys = ItemFormatter.FIXED_KEYS.copy()
             for archive_info in manifest.list_archive_infos(sort_by='ts'):
                 if args.prefix and not archive_info.name.startswith(args.prefix):
                     continue
-                if args.short:
+                if args.format:
+                    archive_info_keys.update({'name': archive_info.name,
+                                              'id': bin_to_hex(archive_info.id),
+                                              'time': format_time(to_localtime(archive_info.ts))})
+                    write(safe_encode(format_line(args.format, archive_info_keys)))
+                elif args.short:
                     print(archive_info.name)
                 else:
                     print(format_archive(archive_info))
```

Comment:
please implement this in a similar way as it was done for the item listing:
- implement a ArchiveFormatter class
- process short and standard format by just using the respective format string